### ISSUE TITLE
Appending Stage name to the Endpoint causes AccessDeniedException when using APIGW Custom Domains

### ISSUE THREAD

**Issue by n1te1337** [michalkvasnicak/aws-lambda-graphql#112]
API Gateway Custom Domains map to an Endpoint **and** Stage name ([link](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-custom-domains.html)), meaning:
default invoke uri: `api-id.execute-api.region.amazonaws.com/stage`
becomes: `example.com`

Currently, the stage name is always appended to the endpoint regardless of the domain type:
https://github.com/michalkvasnicak/aws-lambda-graphql/blob/47cd7fe2cfdd5fd65c5bdadfeddf011307cb9ee0/packages/aws-lambda-graphql/src/Server.ts#L278
https://github.com/michalkvasnicak/aws-lambda-graphql/blob/47cd7fe2cfdd5fd65c5bdadfeddf011307cb9ee0/packages/aws-lambda-graphql/src/helpers/extractEndpointFromEvent.ts#L6

This causes `AccessDeniedException` when the Lambda tries to send updates back to the clients as the `${endpoint}/${stage}` API doesn't exist.

This behaviour should be configurable so that custom domains can be used with the library.

**Comment by michalkvasnicak** [comment]
@n1te1337 Please feel free to open a PR for this. I don't have time at the moment to implement this, so if it's blocking you I'm happy to accept a PR.

**Comment by n1te1337** [comment]
@michalkvasnicak Sure, I can take a stab at it. 

Any preference where the flag should live? Potentially the `subscriptionOptions` object might be a good home for it:
https://github.com/michalkvasnicak/aws-lambda-graphql/blob/master/packages/aws-lambda-graphql/src/Server.ts#L61-L121

**Comment by michalkvasnicak** [comment]
@n1te1337 Is it possible to automatically detect from `requestContext` ?

**Comment by michalkvasnicak** [comment]
Otherwise I'm ok with your suggestion too 👍

**Comment by n1te1337** [comment]
@michalkvasnicak the only way to detect it from  the`requestContext` would be to check if the `domainName` field has a `*.amazonaws.com` suffix. Thoughts?

**Comment by michalkvasnicak** [comment]
Ok, I don't like detecting it that way, I think for now introducing new option to `subscriptionOptions` should be sufficient. Later we can extend the option with some function that can be used to detect endpoint. Or maybe you can implement it right away as `subscriptionOptions.connectionEndpoint?: string | (event: APIGatewayWebSocketEvent) => string`.

Could you please document it too?

**Comment by n1te1337** [comment]
@michalkvasnicak I've added it as a `string` for now. Let me know if you have any feedback.

**Comment by cjnoname** [comment]
@n1te1337 

Hey mate, thanks for your great work. 

I have added this new option to my code, it works great with the WebSocketHandler, the error was gone.
However it's not working on EventHandler.
I got the same error on Cloudwatch log on EventHandler: 
NotFoundException: No method found matching route dev/@connections/AJ822cXGywMCGBw= for http method POST.

Anyone can give me some suggestions?
Thanks @michalkvasnicak @n1te1337
